ai/storage.py

```diff
@@ -124,25 +127,28 @@ async def release(context, stored_drone):
     '''
     Relase a drone from storage on command.
     '''
-
-    LOGGER.debug('Message is valid for release.')
-    if not roles.has_role(context.author, roles.HIVE_MXTRESS):
+    if not roles.has_role(context.author, roles.HIVE_MXTRESS) or context.channel.name != STORAGE_FACILITY:
         return False
 
+    if type(store_drone) is not discord.Member:
+        release_id = stored_drone
+        stored_drone = convert_id_to_member(context.guild, stored_drone)
+
+    if stored_drone is None:
+        return
+
     stored_role = get(context.guild.roles, name=roles.STORED)
     # find stored drone
-    member = stored_drone
-    to_release_id = get_id(member.display_name)
-    storage_to_release = fetch_storage_by_target_id(to_release_id)
-    if storage_to_release is not None:
-        # TODO: does this really return None or throw an error?
-        drone = fetch_drone_with_drone_id(storage_to_release.target_id)
-        member = context.guild.get_member(drone.id)
-        await member.remove_roles(stored_role)
-        await member.add_roles(*get_roles_for_names(context.guild, storage_to_release.roles.split('|')))
-        delete_storage(storage_to_release.id)
+    
+    stored_drone_data = fetch_storage_by_target_id(release_id)
```

Comment:
`release_id` is of type Member but `fetch_storage_by_target_id` takes parameter of type string. Does this work?